tests/test-cast-to-overflow.c

```diff
@@ -39,55 +39,234 @@ typedef struct
   int line;
   int round_mode;
   _Decimal128 set;
+  long double expect;
+} d128_type;
+
+typedef struct
+{
+  int line;
+  int round_mode;
+  _Decimal64 set;
   double expect;
-} d_type;
+} d64_type;
+
+typedef struct
+{
+  int line;
+  int round_mode;
+  _Decimal32 set;
+  float expect;
+} d32_type;
+
+/* Decimal 128 to Binary Double 128.  */
+static const d128_type d128to128[] = {
+ {__LINE__, FE_TONEAREST, 1e6000DL, INFINITY},
+ {__LINE__, FE_TONEAREST, -1e6000DL, -INFINITY},
+ {__LINE__, FE_TONEAREST, 1e4933DL, INFINITY},
+ {__LINE__, FE_TONEAREST, -1e4933DL, -INFINITY},
+#if __LDBL_MANT_DIG__ == 106
+ {__LINE__, FE_TONEAREST, 1.797693134862316e308DL, INFINITY},
+ {__LINE__, FE_TONEAREST, -1.797693134862316e308DL, -INFINITY},
+#else
+ {__LINE__, FE_TONEAREST, 1.189731495357232e4932DL, INFINITY},
+ {__LINE__, FE_TONEAREST, -1.189731495357232e4932DL, -INFINITY},
```

Comment:
I understand that these numbers are LDBL_MAX and -LDBL_MAX for IBM long double and IEEE long double (binary128), respectively, but I wonder if there's a way to express them in a way that makes this fact more clear...

Maybe with macros:
```
#if __LDBL_MANT_DIG__ == 106
# define LDBL_MAX_IN_DECIMAL 1.797693134862316e308DL
#else
# define LDBL_MAX_IN_DECIMAL 1.189731495357232e4932DL
#endif
(etc.)
```

Or maybe with an explicit cast (would this achieve the desired goal of this test?):
```
 {__LINE__, FE_TONEAREST, (_Decimal128) LDBL_MAX, INFINITY},
 {__LINE__, FE_TONEAREST, (_Decimal128) -LDBL_MAX, -INFINITY},
```
(then there would be no need to test for `__LDBL_MANT_DIG__`)

Reply:
This comment, as well as the other one, shouldn't necessarily block this pull request, it's intended as a question (i.e.: are these constants indeed LDBL_MAX, FLT_MAX, etc. expressed as decimal constants? or is there some detail about them?).

Reply:
The numbers are not LDBL_MAX. Instead they are a little bit greater than LDBL_MAX.
I've defined some macros for those values to get rid of "#if __LDBL_MANT_DIG__"
and to point out, that those values are not LDBL_MAX.
See the further commit "Make test-cast-to-overflow.c more readable." on my branch. The original commit is unchanged.